tests/integrationv2/test_key_update.py

```diff
@@ -53,7 +53,7 @@ def test_s2n_server_key_update(managed_process, cipher, provider, other_provider
     server_options.data_to_send = [server_data]
 
     server = managed_process(
-        S2N, server_options, send_marker=[str(client_data)], timeout=30
+        S2N, server_options, send_marker=strip_string_of_bytes(str(client_data)), timeout=30
```

Comment:
nit: `strip_string_of_bytes` seems like a strange and potentially unreliable way to convert a `bytes` object to a string. I wonder if it's worth not pulling it out of `test_record_padding` and converting it in a more reliable way for other tests?
```suggestion
        S2N, server_options, send_marker=client_data.decode(), timeout=30
```

Reply:
I have a comment about this in the [renegotiate test](https://github.com/aws/s2n-tls/blob/main/tests/integrationv2/test_renegotiate.py#L98-L102) because our framework itself is so inconsistent with its conversions that it causes problems.

Your handling of the string in _communicate is probably contributing to the weirdness. Can you maybe prepend the old bytes before converting to data_str? I'd try not to do any conversions there if you don't need to.

Reply:
I _am_ preprending the old bytes before converting to a string. The problem with the keyupdate test is that str(bytes) isn't a good way of converting to a string because it leaves the 'b' marker in the string. Anyways, I can switch to doing bytes.decode() here, but I don't think it has much to do with my handling of strings in _communicate.